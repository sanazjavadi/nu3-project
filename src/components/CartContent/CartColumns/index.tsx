const CartColumns = () => {
  return (
    <thead
      data-testid="cart-columns-container"
      className="text-xs text-gray-700 uppercase border border-b-300 bg-gray-50 dark:bg-gray-700 dark:text-gray-400"
    >
      <tr>
        <th data-testid="image-column-header" scope="col" className="px-6 py-3">
          image
        </th>
        <th data-testid="item-column-header" scope="col" className="px-6 py-3">
          item
        </th>
        <th data-testid="price-column-header" scope="col" className="px-6 py-3">
          price
        </th>
        <th
          data-testid="quantity-column-header"
          scope="col"
          className="px-6 py-3"
        >
          quantity
        </th>
        <th
          data-testid="subtotal-column-header"
          scope="col"
          className="px-6 py-3"
        >
          subtotal
        </th>
      </tr>
    </thead>
  );
};

export default CartColumns;
