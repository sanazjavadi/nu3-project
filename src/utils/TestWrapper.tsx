import { configureStore, Store } from "@reduxjs/toolkit";
import { MemoryRouter } from "react-router-dom";
import { ComponentType, FunctionComponent } from "react";
import { Provider } from "react-redux";
import reducer from "../reducers";

export const makeStore = (): Store => {
  return configureStore({ reducer });
};

const wrapComponent = (
  Component: ComponentType | FunctionComponent | any,
  store: Store | null = null,
  props = {}
): JSX.Element => {
  return (
    <Provider store={store || makeStore()}>
      <MemoryRouter>
        <Component {...props} />
      </MemoryRouter>
    </Provider>
  );
};

export default wrapComponent;
