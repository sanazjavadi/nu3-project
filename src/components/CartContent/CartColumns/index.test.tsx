import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import CartColumns from "./";
import TestWrapper, { makeStore } from "../../../utils/TestWrapper";

const store = makeStore();

describe("CartColumns Tests Cases", () => {
  test("CartColumns: Renders", () => {
    render(TestWrapper(CartColumns, store));
    const cartTableHeaderElement = screen.getByTestId("cart-columns-container");
    expect(cartTableHeaderElement).toBeInTheDocument();
  });

  test("CartColumns: Renders Image Column Header", () => {
    render(TestWrapper(CartColumns, store));
    const imageColumnHeaderElement = screen.getByTestId("image-column-header");
    expect(imageColumnHeaderElement).toBeInTheDocument();
  });

  test("CartColumns: Renders Item Column Header", () => {
    render(TestWrapper(CartColumns, store));
    const itemColumnHeaderElement = screen.getByTestId("item-column-header");
    expect(itemColumnHeaderElement).toBeInTheDocument();
  });

  test("CartColumns: Renders Price Column Header", () => {
    render(TestWrapper(CartColumns, store));
    const priceColumnHeaderElement = screen.getByTestId("price-column-header");
    expect(priceColumnHeaderElement).toBeInTheDocument();
  });

  test("CartColumns: Renders Quantity Column Header", () => {
    render(TestWrapper(CartColumns, store));
    const quantityColumnHeaderElement = screen.getByTestId(
      "quantity-column-header"
    );
    expect(quantityColumnHeaderElement).toBeInTheDocument();
  });

  test("CartColumns: Renders Subtotal Column Header", () => {
    render(TestWrapper(CartColumns, store));
    const subtotalColumnHeaderElement = screen.getByTestId(
      "subtotal-column-header"
    );
    expect(subtotalColumnHeaderElement).toBeInTheDocument();
  });
});

test("CartColumns: Snapshot", () => {
  const component = renderer.create(TestWrapper(CartColumns, store));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
