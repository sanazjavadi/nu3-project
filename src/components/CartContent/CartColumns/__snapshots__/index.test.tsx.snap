// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartColumns: Snapshot 1`] = `
<thead
  className="text-xs text-gray-700 uppercase border border-b-300 bg-gray-50 dark:bg-gray-700 dark:text-gray-400"
  data-testid="cart-columns-container"
>
  <tr>
    <th
      className="px-6 py-3"
      data-testid="image-column-header"
      scope="col"
    >
      image
    </th>
    <th
      className="px-6 py-3"
      data-testid="item-column-header"
      scope="col"
    >
      item
    </th>
    <th
      className="px-6 py-3"
      data-testid="price-column-header"
      scope="col"
    >
      price
    </th>
    <th
      className="px-6 py-3"
      data-testid="quantity-column-header"
      scope="col"
    >
      quantity
    </th>
    <th
      className="px-6 py-3"
      data-testid="subtotal-column-header"
      scope="col"
    >
      subtotal
    </th>
  </tr>
</thead>
`;
